### ISSUE TITLE
Failure to import working conda env (arcgis pro): due to incompatible packages

### ISSUE THREAD

**Issue by maphew** [prefix-dev/pixi#1876]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
pushd D:\testing
 C:\ArcGIS\Pro\bin\Python\condabin\conda.bat env export -n arcgispro-py3 > arcgispro-py3-environment.yml

pixi --version
pixi 0.27.1

pixi init --import arcgispro-py3-environment.yml
  x failed to solve the conda requirements of 'default' 'win-64'
  `-> Cannot solve the request because of: The following packages are incompatible
      └─ zstd ==1.5.5 1 cannot be installed because there are no viable options:
         └─ zstd 1.5.5 is excluded because due to strict channel priority not using this option from: 'https://conda.anaconda.org/esri/'
```
[arcgispro-py3-environment.yml.txt](https://github.com/user-attachments/files/16696143/arcgispro-py3-environment.yml.txt)
[pixi.toml.txt](https://github.com/user-attachments/files/16696157/pixi.toml.txt)

The current reported depencency list according to conda recipe:

```
pixi search arcpy
Using channels from project (arcgispro-py3): conda-forge, main, r, msys2, esri

arcpy py311_arcgispro_52636
---------------------------

Name                arcpy
Version             3.3
Build               py311_arcgispro_52636
Size                74150
License             Not found.
Subdir              win-64
File Name           arcpy-3.3-py311_arcgispro_52636.conda
URL                 https://conda.anaconda.org/esri/win-64/arcpy-3.3-py311_arcgispro_52636.conda
MD5                 5cc17d45855386e188904373e5273763
SHA256              32fc2c747f95847743d8f767bf86da8184d95998a590b16b81e1f2dda3bff5e5

Dependencies:
 - appdirs
 - arcgis 2.3.*
 - arcpy-base 3.3 py311_52636
 - azure-storage-blob
 - black
 - chardet
 - cryptography
 - flake8
 - freetype >=2.13.2
 - jdcal
 - matplotlib
 - matplotlib-base
 - networkx
 - ntlm-auth
 - olefile
 - pillow >=10.2.0
 - pro_notebook_integration 3.3.*
 - protobuf 4.21.12.*
 - psutil
 - pyodbc
 - pytest >=5.4.1
 - python >=3.11,<3.12.0a0
 - pytz
 - regex
 - saspy
 - scipy >=1.5.2
 - seaborn
 - sqlalchemy 1.4.*
 - statsmodels
 - swat
 - sympy
 - testpath
 - tqdm
 - tzdata
 - xarray >=0.16
```

### Issue description

Requires a machine with ArcGIS Pro v3.3. installed.

The conda environment works so I don't understand why import fails. Perhaps the published `.conda` recipe differs from the on-disk situation? Suggestions on how to compare the two welcome.

### Expected behavior

That pixi would be able to replicate an existing and working conda environment.

**Comment by olivier-lacroix** [comment]
Based on the error message, maybe try by [disabling channel strict priority](https://pixi.sh/v0.27.1/reference/project_configuration/#channel-priority-optional). This will make pixi behave like conda's default

**Comment by maphew** [comment]
Thanks Olivier! Disabling channel priority is the main necessary change.

I also needed to generate an _environment.yml_ from a pristine Pro v3.3. install. Apparently the one I started with had some extra pypi packages installed. For completeness sake, the pristine yml is attached.

[arcgispro-py3-v3-3-1-environment.yml.txt](https://github.com/user-attachments/files/16752622/arcgispro-py3-v3-3-1-environment.yml.txt)
